api/spec/v1.yaml

```diff
@@ -1397,10 +1531,37 @@ components:
     RuntimeToken:
       type: object
       properties:
-        token_addr:
+        contract_addr:
           type: string
           description: The Oasis address of this token's contract.
           example: 'oasis1qp2hssandc7dekjdr6ygmtzt783k3gn38uupdeys'
+        name:
+          type: string
+          description: Name of the token, as provided by token contract's `name()` method.
+          example: Uniswap
+        symbol:
+          type: string
+          description: Symbol of the token, as provided by token contract's `symbol()` method.
+          example: UNI
+        decimals:
+          type: integer
+          description: |
+            The number of least significant digits in base units that should be displayed as
+            decimals when displaying tokens. `tokens = base_units / (10**decimals)`.
+            Affects display only. Often equals 18, to match ETH.
+          example: 18
+        type:
+          type: string
+          enum: *evm_token_types
+          nullable: true
+          description: |
+            The heuristically determined interface that the token contract implements.
+            A less specialized variant of the token might be detected; for example, an
+            ERC-1363 token might be labeled as ERC-20 here. If the type cannot be
+            detected or is not supported, this field will be null/absent.
+        total_supply:
+          type: string
+          description: The total number of base units available.
```

Comment:
```suggestion
        total_supply:
          type: string
          description: The total number of base units available, as provided by the token contract's `totalSupply()` method.
```

Reply:
Not all the time. We'll query it from time to time but we can't promise to be up to date. (Unlike with symbol() and name(), which are supposed to be static.) So I imagine we'll do a combo of calling totalSupply() and doing dead reckoning via events emitted by [_mint()](https://docs.openzeppelin.com/contracts/2.x/api/token/erc20#ERC20-_mint-address-uint256-) and _burn.

Reply:
on a tangent,

> symbol() and name(), which are supposed to be static

is that stated? like we won't really support letting them change, but we can commence feeling good if spec says they won't change

Reply:
It doesn't look like it's promised/enforced in any way: https://eips.ethereum.org/EIPS/eip-20
It's still reasonable for us to behave unreasonably towards a hypothetica tokenwith an un unresonable `symbol()` impl, IMO.